setup.py

```diff
@@ -61,7 +61,7 @@ def get_extras_require() -> Dict[str, List[str]]:
             "mlflow",
         ],
         # TODO(hvy): Unpin `sphinx` version after https://github.com/sphinx-doc/sphinx/issues/7807.
-        "document": ["sphinx>=3.0.0,!=3.1.0,!=3.1.1", "sphinx_rtd_theme"],
+        "document": ["sphinx>=3.0.0,!=3.1.0,!=3.1.1", "sphinx_rtd_theme>=0.5.0"],
```

Comment:
I have a question about this version restriction. Is it necessary for us to update the `sphinx_rtd_theme` version to show the type hints correctly? Or, is it optional?

I think it's acceptable to use the latest sphinx, but I'm a bit concerned about removing the version restriction set by ReadTheDocs. I briefly checked the issues in the `sphinx_rtd_theme` repository, and I found some report which claims `v0.5.0` unintended changes some layouts (e.g., https://github.com/readthedocs/sphinx_rtd_theme/issues/949). I don't think problems occur in Optuna reference as far as I checked https://harupy-optuna.readthedocs.io/en/latest/reference/study.html.

Reply:
> Is it necessary for us to update the sphinx_rtd_theme version to show the type hints correctly?

I thought it **was**, but it actually **isn't**. So let's stick to `sphinx_rtd_theme<0.5`.

Reply:
Thank you for your reply. I support your decision to focus only on the Python version.
Could you update the PR title too?

Reply:
Sure. I'll update the title too.